### DISCUSSION TITLE
Add token filter to prevent spilling private API keys

### DISCUSSION THREAD

**Discussion by Leone25** [community/community#17561] [Other Feature Feedback, Questions, & Ideas]
Hello,

I just have a random idea that might help prevent future API key spillage when committing code.
I think it'd be nice for Github to have a page in the account settings where one can put all their private keys and when committing new code Github is gonna scan it for matches and report an error if any, preventing the commit from going public. A bit like it already does with discord tokens (they are actually disabled, but that requires integration with other apps and it's not really feasible).

I hope this gets implemented at some point.

Cheers

Rico

**Comment by judicaelandria** [comment]
I think this is a great idea too! if has the workflow as the website deployment for secret variable like vercel
